.github/workflows/ci.yml

```diff
@@ -40,42 +45,47 @@ jobs:
         run: cargo fmt --all -- --check
 
   test_release:
-    runs-on: ubuntu-24.04
+    runs-on: ['self-hosted', 'linux', 'x64', '4xlarge']
     name: Test in release mode
     strategy:
       matrix:
         cargo-args: ['', '--features fixed-rows-to-discard']
+      fail-fast: false
     env:
       # Run all tests with multicore-SDR enabled.
       FIL_PROOFS_USE_MULTICORE_SDR: true
     steps:
       - uses: actions/checkout@v4
       - name: Install required packages
-        run: sudo apt install --no-install-recommends --yes libhwloc-dev ocl-icd-opencl-dev
+        run: |
+          sudo apt-get update
+          sudo apt-get install --no-install-recommends --yes libhwloc-dev ocl-icd-opencl-dev
 
       - name: Download the proof params
         uses: ./.github/actions/proof-params-download
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
 
+      - uses: dtolnay/rust-toolchain@21dc36fb71dd22e3317045c0c31a3f4249868b17
+        with:
+          toolchain: 1.83
+
```

Comment:
I recommend installing Rust on the self hosted runner via [rustup](https://rustup.rs). This way you wouldn't need this. The benefit is that then invocations of `cargo` would automatically install the Rust version specified in the `rust-toolchain` file that is in the root directory of this repo.

This would then also mean that you wouldn't need the CI file, whenever you update the version in `rust-toolchain`.

Reply:
That's a good point. I'll add a TODO for this in our backlog. To be addressed when we rebuild the AMI for the machines.

I also want to install the GPU drivers in the AMI as well so that we wouldn't have to do that every time a job runs.